src/components/Fieldset/Fieldset.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames/bind';
+import styles from './Fieldset.scss';
+const cx = classnames.bind( styles );
+
+const Fieldset = ( {
+	autoComplete,
+	defaultValue,
+	describedBy,
+	disabled,
+	isMultiline,
+	inputRef,
+	id,
+	label,
+	maxLength,
+	placeholder,
+	onBlur,
+	onChange,
+	onClick,
+	onFocus,
+	onInvalid,
+	pattern,
+	readOnly,
+	required,
+	type,
+} ) => {
+	// ClassNames used by both <input> and <textarea>.
+	const sharedClassNames = {
+		disabled,
+		readOnly,
+		[type]: type,
+	};
+
+	const requiredField = label && required;
+
+	return (
+		<fieldset className={styles.container}>
+			<label className={`${styles.label} ${styles.requiredLabel}`}>
+				{label}
+				{requiredField && <span className={styles.requiredAsterisk}> * </span>}
+				{isMultiline &&
+					<textarea
+						className={cx( 'textarea', { ...sharedClassNames } )}
+						rows={6}
+						aria-describedby={describedBy}
+						autocomplete={autoComplete}
+						defaultValue={defaultValue}
+						describedBy={describedBy}
+						disabled={disabled}
+						id={id}
+						maxLength={maxLength}
+						placeholder={placeholder}
+						onBlur={onBlur}
+						onChange={onChange}
+						onClick={onClick}
+						onFocus={onFocus}
+						onInvalid={onInvalid}
+						readOnly={readOnly}
+						ref={inputRef}
+						required={required}
+					/>
+				}
+
+				{!isMultiline &&
+					<input
+						className={cx( 'input', { ...sharedClassNames } )}
+						type={type}
+						aria-describedby={describedBy}
+						autocomplete={autoComplete}
+						defaultValue={defaultValue}
+						describedBy={describedBy}
+						disabled={disabled}
+						id={id}
+						maxLength={maxLength}
+						placeholder={placeholder}
+						readOnly={readOnly}
+						onBlur={onBlur}
+						onChange={onChange}
+						onClick={onClick}
+						onFocus={onFocus}
+						onInvalid={onInvalid}
+						pattern={pattern}
+						readOnly={readOnly}
+						ref={inputRef}
+						required={required}
+						type={type}
+					/>
+				}
+			</label>
+		</fieldset>
+	);
+};
+
+Fieldset.propTypes = {
+	/**
+	 * Input tag prop; boolean to accept autoCompletion or not.
+	 */
+	autoComplete: PropTypes.string,
+	/**
+	 * Input tag prop; sets default value.
+	 */
+	defaultValue: PropTypes.string,
+	/**
+	 * Input tag prop; used with aria-describedby attribute to indicate the ID of a linked description element.
+	 */
+	describedBy: PropTypes.string,
+	/**
+	 * Input tag prop; sets whether the input is disabled from changing values or not.
+	 */
+	disabled: PropTypes.bool,
+	/**
+	 * Used to identify the element.
+	 */
+	id: PropTypes.string,
+	/**
+	 * Used in the event of forwarding a ref to the input tag.
+	 */
+	inputRef: PropTypes.object,
+	/**
+	 * Boolean that determines whether component is a `textarea` or `input` tag.
+	 */
+	isMultiline: PropTypes.bool.isRequired,
+	/**
+	 * Text to be used in `label` tag.
```

Comment:
Help me understand what value I should provide here without going into the inner workings of the component. E.g., 'Text displayed to the user above the input'